Source/Particles/WarpXParticleContainer.H

```diff
@@ -332,7 +332,7 @@ public:
     /**
      * \brief Virtual method to resample the species. Overriden by PhysicalParticleContainer.
      */
-    virtual void resample (const Resampling& /*resampler*/, const int /*timestep*/) {};
+    virtual void resample (const Resampling& /*resampler*/, const int /*timestep*/) {}
```

Comment:
@NeilZaim generally, if we want this to be overwritten and implemented in a deriving class, we can even make this purely virtual in the interface class:
```suggestion
    virtual void resample (const Resampling& /*resampler*/, const int /*timestep*/) = 0;
```

Reply:
Sorry, that implies that we implement (and empty body) for this function in all derived classes. `LaserParticleContainer` does not have that method implemented yet.